File: Python/record.py
import datetime

class Record:
    key: str
    value: str
    timestamp: datetime.datetime
    valid: bool

    def __init__(self, key: str = "", value: str = ""):
        self.key = key
        self.value = value
        self.timestamp = datetime.datetime.now()
        self.valid = True

    def invalidate(self):
        self.valid = False

    def __str__(self):
        return "{} {} {}".format(self.key, self.value, self.timestamp)

File: Ruby/record.rb
require 'date'

class Record
  @key
  @value
  @timestamp

  def initialize(key, value)
    @key = key
    @value = value
    @timestamp = DateTime.now
  end

  def key
    @key
  end

  def value
    @value
  end

  def timestamp
    @timestamp
  end

  def to_s
    "#{@key} #{@value} #{@timestamp}"
  end
end
